### PR TITLE
Support perplexity evals with context and tensor parallelism

### PR DESCRIPTION
**Pull request by tyler-romero**
## Summary
- Support PPL evals with both CP and TP in `TransformerTrainModule.eval_batch()`.
- When CP or TP is active, shard `label_mask` along the sequence dimension to match the sharded `ce_loss` shape. CP shards first via the load balancer, then TP shards further via `torch.chunk`.
- When TP is active, set `return_logits=False` and unwrap the DTensor `ce_loss` with `get_local_tensor()`.

## Test plan
- [x] Run a training job with CP and/or TP enabled with `LMEvaluator` configured — verify PPL metrics are reported
- [x] Downstream evaluators that require full logits still fail naturally with CP or TP

### CODE REVIEW

**Comment by tyler-romero** [comment]
TP: https://beaker.org/orgs/ai2/workspaces/oe-t-ladder/work/01KJ66NSZFVRPC9G1P92GWTA7R

**Comment by tyler-romero** [comment]
cp: https://beaker.org/orgs/ai2/workspaces/oe-t-ladder/work/01KJ626B6VGK8ZV5AGCG3JBGDG

File: src/olmo_core/nn/transformer/model.py [hunk 1]
```diff
@@ -240,8 +240,11 @@ def get_rope_buffers(
             device = self.device
         rope_buffers = {}
         for key, block in self.blocks.items():
-            rope = cast(Optional[RotaryEmbeddingBase], block.attention.rope)  # type: ignore
-            rope_buffers[int(key)] = None if rope is None else rope.get_buffers(seq_len, device)
+            if isinstance(block.attention, (Attention, FusedAttention)):
```
Comment (tyler-romero) [inline src/olmo_core/nn/transformer/model.py:243]: unrelated cp test fix
